File: leetcode/move_zeroes_283.py
from typing import List
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        j = 0
        # while i < len(nums):
        #     if nums[i] != 0:
        #         nums[j] = nums[i]
        #         j+=1
        #     i+=1

        j = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[j] = nums[i]
                j+=1


        for i in range(j, len(nums)):
            nums[i] = 0

        return nums

a = Solution().moveZeroes([0,0,1])
print(a)

File: leetcode/week1/move_zeroes_283.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == 0) {
      var move = i+1
      console.log(nums[i]);
      while (move < nums.length){
        if (nums[move] != 0) {
          nums[i] = nums[move];
          nums[move] = 0;
          break;
        }
        move += 1;
      };
    };
  };
  return nums;
};